File: backend/internal/domain/chef.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Chef struct {
	ChefID    uuid.UUID `json:"chef_id"`
	HotelID   uuid.UUID `json:"hotel_id"`
	Name      string    `json:"name"`
	BioEN     string    `json:"bio_en"`
	BioAM     string    `json:"bio_am"`
	ImageURL  string    `json:"image_url"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type CreateChefRequest struct {
	HotelID  uuid.UUID `json:"hotel_id" binding:"required"`
	Name     string    `json:"name" binding:"required"`
	BioEN    string    `json:"bio_en"`
	BioAM    string    `json:"bio_am"`
	ImageURL string    `json:"image_url"`
}

type UpdateChefRequest struct {
	Name     *string `json:"name"`
	BioEN    *string `json:"bio_en"`
	BioAM    *string `json:"bio_am"`
	ImageURL *string `json:"image_url"`
}

File: frontend/src/types/chef.ts
export interface Chef {
    chef_id: string;
    hotel_id: string;
    name: string;
    bio_en?: string;
    bio_am?: string;
    image_url?: string;
    created_at: string;
    updated_at: string;
}

export interface CreateChefRequest {
    hotel_id: string;
    name: string;
    bio_en?: string;
    bio_am?: string;
    image_url?: string;
}

export interface UpdateChefRequest {
    name?: string;
    bio_en?: string;
    bio_am?: string;
    image_url?: string;
}
